spec/services/case_validate_rejected_offender_sar_service_spec.rb

```diff
@@ -0,0 +1,56 @@
+require "rails_helper"
+
+describe CaseValidateRejectedOffenderSARService do
+  describe "#call" do
+    let(:team)          { find_or_create :team_branston }
+    let(:user)          { find_or_create :branston_user }
+    let(:kase)          { create :offender_sar_case, :rejected, received_date: 1.day.ago.to_date }
+    let(:state_machine) { double ConfigurableStateMachine::Machine,  validate_rejected_case!: true } # rubocop:disable RSpec/VerifiedDoubles
+    let(:service)       do
+      described_class.new(user:,
+                          kase:,
+                          params:)
+    end
+
+    before do
+      allow(kase).to receive(:state_machine).and_return(state_machine)
+    end
+
+    context "when setting a valid date" do
+      let(:params)        { { received_date: Time.zone.today } }
+
+      it "changes the received date on the case" do
+        service.call
+        expect(kase.received_date).to eq Time.zone.today
+      end
+
+      it "transitions the cases state" do
+        expect(state_machine).to receive(:validate_rejected_case!).with({ acting_user: user,
+                                                                          acting_team: team,
+                                                                          message: nil })
+        service.call
+        expect(service.result).to eq :ok
+      end
+
+      it "removes the 'R' from the case number" do
+        service.call
+        expect(kase.number[0]).not_to eq "R"
+      end
+
+      it "sets results to :ok" do
+        service.call
+        expect(service.result).to eq :ok
+      end
+    end
+
+    context "when anything fails in the transaction" do
+      let(:params) { { received_date: "" } }
+
+      it "raises an error when it saves" do
+        allow(kase).to receive(:save).and_raise(RuntimeError)
```

Comment:
Calling `save` won't raise an error. 
It would be better to use invalid params rather than mocking the method so you are fully testing what the method does.